### PR TITLE
Py3 raet log.warn

### PR DESCRIPTION
**Pull request by FirefighterBlu3**
logging.warn() is deprecated, change from logging.warn to logging.warning() and a few quoting style changes for consistency throughout salt

### CODE REVIEW

File: salt/grains/core.py [hunk 1]
```diff
@@ -656,8 +656,8 @@ def _virtual(osdata):
                 grains['virtual'] = 'kvm'
     else:
         if osdata['kernel'] in skip_cmds:
-            log.warn(
-                'The tools \'dmidecode\' and \'lspci\' failed to '
+            log.warning(
+                "The tools 'dmidecode' and 'lspci' failed to '
```
Comment (cachedout) [inline salt/grains/core.py]: This is an unmatched double-quote.

**Comment by rallytime** [comment]
@FirefighterBlu3 Did you see the comment from @cachedout above? It looks like this pull request also needs to be rebased. If you can get that fixed up, we can take another look at this and get this change in.

**Comment by FirefighterBlu3** [comment]
yes, i'll try to get to that later today. thank you -d

**Comment by FirefighterBlu3** [comment]
ok, fixed for the unmatched double-quote, rebasing shortly

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by FirefighterBlu3** [comment]
there are 5,026 lines in salt files that use the form of '...\'..' strings which is discouraged in PEP8 in favor of the following string style; "....'..". i've written a short parser to rewrite these and it appears to be operating with 100% success. what're the feelings regarding doing a seemingly large style change like this?

bear in mind that there are about 309,879 lines with strings that don't use this form so these 5K only represent about 1% of the salt project.

**Comment by cachedout** [comment]
Hi @FirefighterBlu3 

We're unlikely to merge such a large change purely in the name of style. We've really been bitten by those in the past. While I understand your PEP8 concern, we do maintain our own subset of style-guidelines, one of which encourages the use of single-quotes over double-quotes: https://docs.saltstack.com/en/latest/topics/development/conventions/style.html#single-quotes

**Comment by cachedout** [comment]
I tested this against Python 2.6 and 2.7 and it appears to work fine.

**Comment by FirefighterBlu3** [comment]
right, i was questioning however since the majority of quoting in salt is not escaping single quotes and escaping them does make it difficult to read. i strongly prefer single quotes myself. thank you for the merge!

**Pull request merged**